### ISSUE TITLE
Support Yi-VL-6B/34B

### ISSUE THREAD

**Issue by paulcx** [sgl-project/sglang#91]
The Yi-VL adopts llava but with silightly different in weights and inference. see [disscusion](https://huggingface.co/01-ai/Yi-VL-34B/discussions/3)

hf repo:
https://huggingface.co/01-ai/Yi-VL-6B
https://huggingface.co/01-ai/Yi-VL-34B

**Comment by merrymercy** [comment]
@paulcx It should be easy to do. Could you help us support it?

The Llava implementation is here https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/llava.py
The single batch example is https://github.com/sgl-project/sglang/blob/main/examples/quick_start/srt_example_llava.py
The batch example is https://github.com/sgl-project/sglang/tree/main/benchmark/llava_bench

**Comment by paulcx** [comment]
I guess something I can quickly fix by add a specific mlp2x_gelu_Norm projector class (not good solution at all):
```
class YiLlavaMultiModalProjector(nn.Module):
    def __init__(self, config):
        super().__init__()

        self.projector = nn.Sequential(
            nn.Linear(config.mm_hidden_size, config.hidden_size),
            nn.GELU(),
            nn.LayerNorm(config.hidden_size),
            nn.Linear(config.hidden_size, config.hidden_size),
            nn.LayerNorm(config.hidden_size),
        )

    def forward(self, image_features):
        return self.projector(image_features)
```

and then add codes:
```
self.multi_modal_projector = (
            YiLlavaMultiModalProjector(config)
            if config.mm_projector_type == "mlp2x_gelu_Norm"
            else LlavaMultiModalProjector(config)
        )
```
and here
```
if self.config.mm_projector_type == "mlp2x_gelu_Norm":
            projector_weights = {
                "model.mm_projector.0": "multi_modal_projector.projector.0",
                "model.mm_projector.1": "multi_modal_projector.projector.2",
                "model.mm_projector.3": "multi_modal_projector.projector.3",
                "model.mm_projector.4": "multi_modal_projector.projector.4",
            }
        else:
            projector_weights = {
                "model.mm_projector.0": "multi_modal_projector.linear_1",
                "model.mm_projector.2": "multi_modal_projector.linear_2",
            }
```

However, what I did not figure out is [pad_input_ids](https://github.com/sgl-project/sglang/blob/3de54a1b5576efe0a06ce73b856f4aa8f40a8b29/python/sglang/srt/models/llava.py#L35) part. HF implementation is quite different from the original llava code, right? Any suggestion?

Reference:
https://github.com/01-ai/Yi/blob/main/VL/single_inference.py

**Comment by merrymercy** [comment]
If only the weight part is different, it seems the other parts can be directly reused.
Did you find any problems with `pad_input_ids`? Could you share your error messages?
If you see something like `does not have image_token_index`, you can try to update the `config.json` or update the code to handle it correctly.

**Comment by paulcx** [comment]
That's right. Yi uses
```
IMAGE_TOKEN_INDEX = -200
DEFAULT_IMAGE_TOKEN = "<image_placeholder>"
```

**Comment by merrymercy** [comment]
Then there are probably two modifications you need to do

1. Register a new chat template for Yi. https://github.com/sgl-project/sglang/blob/3de54a1b5576efe0a06ce73b856f4aa8f40a8b29/python/sglang/lang/chat_template.py#L12-L18. Set image_token to `"<image_placeholder>"`
2. Update this line accordingly if necessary https://github.com/sgl-project/sglang/blob/3de54a1b5576efe0a06ce73b856f4aa8f40a8b29/python/sglang/srt/models/llava.py#L39

**Comment by merrymercy** [comment]
#112

**Comment by BabyChouSr** [comment]
Closing because it is resolved through #112
